cupy/core/core.pyx

```diff
@@ -4509,62 +4509,23 @@ def _add_scan_blocked_sum_kernel(dtype):
     return module.get_function(name)
 
 
-@util.memoize(for_each_device=True)
-def _nonzero_1d_kernel(src_dtype, index_dtype):
-    name = "nonzero_1d_kernel"
-    src_dtype = _get_typename(src_dtype)
-    index_dtype = _get_typename(index_dtype)
-
-    source = string.Template("""
-    extern "C" __global__ void ${name}(const CArray<${src_dtype}, 1> src,
-        const CArray<${index_dtype}, 1> scaned_index,
-        CArray<${index_dtype}, 1> dst){
-        int thid = blockIdx.x * blockDim.x + threadIdx.x;
-        ptrdiff_t n = src.size();
-        if (thid < n){
-            if (src[thid] != 0){
-                dst[scaned_index[thid] - 1] = thid;
-            }
-        }
-    }
-    """).substitute(name=name, src_dtype=src_dtype, index_dtype=index_dtype)
-    module = compile_with_cache(source)
-    return module.get_function(name)
+cdef _nonzero_kernel_1d = ElementwiseKernel(
+    'T src, S index', 'raw S dst',
+    'if (src != 0) dst[index - 1] = i',
+    '_nonzero_kernel_1d')
```

Comment:
Could you be consistent with that the name of `_nonzero_kernel` is `nonzero_kernel` (no leading underscore)?

Reply:
I fixed.